service/raft/raft_group_registry.cc

```diff
@@ -405,13 +406,15 @@ future<> raft_group_registry::start_server_for_group(raft_server_for_group new_g
     }
 }
 
-void raft_group_registry::abort_server(raft::group_id gid, sstring reason) {
+future<> raft_group_registry::abort_server(raft::group_id gid, sstring reason) {
     // abort_server could be called from on_background_error for group0
     // when the server has not yet been added to _servers.
     // In this case we won't find gid in the if below.
     if (const auto it = _servers.find(gid); it != _servers.end()) {
         ensure_aborted(it->second, std::move(reason));
+        return it->second.aborted->get_future();
     }
+    return make_ready_future<>();
```

Comment:
Here again it `abort_server()` id called twice the second call will return before the server is aborted, no?

Reply:
if we didn't find gid, then the server was already drained and destroyed. If drain_and_abort is in progress, then we'll look at the same abort future.